### PR TITLE
Implement GH-41: Add minimum width gui

### PR DESCRIPTION
**Pull request by mewtaylor**
This implements #41. It adds a new component, `Dropdown`, which is a formatted version of the `Popover` element.

I also moved the `modes.js` file to a `lib` folder, and added `layout-constants` from gui to the lib folder as well. This move from `modes/modes.js` to `lib/modes.js` can be reverted if needed.

@carljbowman and I discussed using the `ContextMenu` for this, however that library is pre-built to pop up wherever there is a click, which is not the desired behavior for this feature.

We could feasibly move the styling for `.mod-menu-item` into its own component on dropdown if we want, but I wasn't sure if that was a good idea or not.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -67,6 +67,7 @@
     "react-intl-redux": "0.6.0",
     "react-popover": "0.5.4",
     "react-redux": "5.0.5",
+    "react-responsive": "^3.0.0",
```
Comment (rschamp) [inline package.json]: We should pin this since it's actively used by the application

File: src/lib/layout-constants.js [hunk 2]
```diff
@@ -0,0 +1,3 @@
+export default {
+    fullSizeMinWidth: 1096
```
Comment (fsih) [inline src/lib/layout-constants.js]: This one says 1096 and the other one says 1095. Is that right?

File: src/components/paint-editor/paint-editor.jsx [hunk 3]
```diff
@@ -342,9 +402,7 @@ PaintEditorComponent.propTypes = {
     canUndo: PropTypes.func.isRequired,
     intl: intlShape,
     name: PropTypes.string,
-    onCopyToClipboard: PropTypes.func.isRequired,
     onGroup: PropTypes.func.isRequired,
-    onPasteFromClipboard: PropTypes.func.isRequired,
```
Comment (fsih) [inline src/components/paint-editor/paint-editor.jsx]: I think you want to add these back too

File: src/components/paint-editor/paint-editor.jsx [hunk 4]
```diff
@@ -248,7 +308,7 @@ class PaintEditorComponent extends React.Component {
                             </InputGroup>
                             <InputGroup className={styles.modModeTools}>
                                 <ModeToolsComponent
-                                    onCopyToClipboard={this.props.onCopyToClipboard}
+                                    onCopyToClipboard={this.props.onCopyToClipboard}
```
Comment (fsih) [inline src/components/paint-editor/paint-editor.jsx]: Nit: there are some extra spaces here

File: src/lib/layout-constants.js (hunk 2 above)
Comment (fsih) [reply src/lib/layout-constants.js]: Also noticed that things stick out a bit before min width UI kicks in, so we might want to do it sooner?
image

**Review by fsih — CHANGES_REQUESTED** [review]

**Comment by fsih** [comment]
If the 1096/1095 difference is intentional, then this PR looks good to me

**Review by fsih — DISMISSED** [review]

**Comment by paulkaplan** [comment]
@fsih I think this requires your re-review.

**Review by fsih — APPROVED** [review]
Sorry! Thanks for the reminder ping

**Pull request merged**